### PR TITLE
Service Bus: Updated 2 templates of Queue and Subscription with auto forwarding

### PR DESCRIPTION
**Pull request by v-Ajnava**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*  Servicebus:  Updated 2 templates of Queue and Subscription with auto forwarding
*  Template : 101-servicebus-queue
* Template : 101-servicebus-topic-subscription

### CODE REVIEW

File: 101-servicebus-queue/azuredeploy.json [hunk 1]
```diff
@@ -1,43 +1,97 @@
 {
-  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "$schema": "http://schema.management.azure.com/schemas/2014-04-01-preview/deploymentTemplate.json#",
```
Comment (bmoore-msft) [inline 101-servicebus-queue/azuredeploy.json]: update the schema verions

File: 101-servicebus-topic-subscription/azuredeploy.json [hunk 2]
```diff
@@ -1,62 +1,102 @@
 {
-  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "$schema": "http://schema.management.azure.com/schemas/2014-04-01-preview/deploymentTemplate.json#",
```
Comment (bmoore-msft) [inline 101-servicebus-topic-subscription/azuredeploy.json]: update the schema version

File: 101-servicebus-topic-subscription/azuredeploy.json [hunk 3]
```diff
@@ -1,62 +1,102 @@
...
+        "description": "Name of the Topic"
+      }
+    },
+    "serviceBusSubscriptionName": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of the Subscription"
+      }
+    },
+    "serviceBusApiVersion": {
```
Comment (bmoore-msft) [inline 101-servicebus-topic-subscription/azuredeploy.json]: remove this and use a literal apiVersion

File: 101-servicebus-topic-subscription/azuredeploy.json [hunk 4]
```diff
@@ -1,62 +1,102 @@
...
   },
   "variables": {
-    "sbVersion": "2015-08-01"
+    "defaultSASKeyName": "RootManageSharedAccessKey",
+    "authRuleResourceId": "[resourceId('Microsoft.ServiceBus/namespaces/authorizationRules', parameters('serviceBusNamespaceName'), variables('defaultSASKeyName'))]"
   },
   "resources": [
     {
-      "apiVersion": "2015-08-01",
+      "apiVersion": "[parameters('serviceBusApiVersion')]",
```
Comment (bmoore-msft) [inline 101-servicebus-topic-subscription/azuredeploy.json]: Use a literal apiVersion so schema validation can be done.

File: 101-servicebus-topic-subscription/azuredeploy.parameters.json [hunk 5]
```diff
@@ -1,15 +1,21 @@
...
+      "value": "GEN-UNIQUE"
+    },
+    "serviceBusTopicName2": {
+      "value": "GEN-UNIQUE"
+    },
+    "serviceBusSubscriptionName": {
+      "value": "GEN-UNIQUE"
+    },
+    "serviceBusApiVersion": {
+      "value": "2017-04-01"
```
Comment (bmoore-msft) [inline 101-servicebus-topic-subscription/azuredeploy.parameters.json]: remove this

File: 101-servicebus-topic-subscription/metadata.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
...
-  "description": "This template creates a Service Bus Namespace and Topic with a Subscription to catch all messages.",
-  "summary": "This template creates a Service Bus Namespace and Topic with a Subscription to catch all messages.",
-  "githubUsername": "crpietschmann",
-  "dateUpdated": "2016-03-28"
-}
+  "itemDisplayName": "Create a Service Bus namespace, topic, and subscription",
+  "description": "This template enables you to deploy a Service Bus Standard namespace, a topic, and a subscription.",
+  "summary": "This template deploys a Service Bus namespace, a topic, and a subscription.",
+  "githubUsername": "v-ajnava",
+  "dateUpdated": "2017-06-22"
```
Comment (bmoore-msft) [inline 101-servicebus-topic-subscription/metadata.json]: update the date

File: 101-servicebus-queue/metadata.json [hunk 7]
```diff
@@ -1,7 +1,7 @@
...
-  "description": "This template creates a Service Bus Namespace and Queue.",
-  "summary": "This template creates a Service Bus Namespace and Queue.",
-  "githubUsername": "crpietschmann",
-  "dateUpdated": "2016-03-30"
-}
+  "itemDisplayName": "Create a Service Bus namespace and queue ",
+  "description": "This template enables you to deploy a Service Bus Standard namespace and a queue.",
+  "summary": "This template creates a Service Bus namespace and 2 queues and sets auto forwarding.",
+  "githubUsername": "v-ajnava",
+  "dateUpdated": "2017-06-22"
```
Comment (bmoore-msft) [inline 101-servicebus-queue/metadata.json]: update the date

File: 101-servicebus-queue/azuredeploy.json (hunk 1 above)
Comment (v-Ajnava) [reply 101-servicebus-queue/azuredeploy.json]: Resolved

File: 101-servicebus-queue/metadata.json (hunk 7 above)
Comment (v-Ajnava) [reply 101-servicebus-queue/metadata.json]: Resolved

File: 101-servicebus-topic-subscription/azuredeploy.json (hunk 2 above)
Comment (v-Ajnava) [reply 101-servicebus-topic-subscription/azuredeploy.json]: Resolved

File: 101-servicebus-topic-subscription/azuredeploy.json (hunk 3 above)
Comment (v-Ajnava) [reply 101-servicebus-topic-subscription/azuredeploy.json]: Resolved

File: 101-servicebus-topic-subscription/azuredeploy.json (hunk 4 above)
Comment (v-Ajnava) [reply 101-servicebus-topic-subscription/azuredeploy.json]: Resolved

File: 101-servicebus-topic-subscription/azuredeploy.parameters.json (hunk 5 above)
Comment (v-Ajnava) [reply 101-servicebus-topic-subscription/azuredeploy.parameters.json]: Resolved

File: 101-servicebus-topic-subscription/metadata.json (hunk 6 above)
Comment (v-Ajnava) [reply 101-servicebus-topic-subscription/metadata.json]: Resolved

**Comment by bmoore-msft** [comment]
@v-Ajnava - check the CI failures

**Comment by v-Ajnava** [comment]
@bmoore-msft  resolved the CI errors

**Comment by v-Ajnava** [comment]
@bmoore-msft can you please review the changes pushed.

**Pull request merged**